### PR TITLE
Tungsten.js Patterns Doc

### PR DESCRIPTION
**Pull request by andrewrota**
# Overview

A general doc on patterns (and anti-patterns) in tungsten.js applications.

### CODE REVIEW

File: docs/patterns.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Events up, methods down
+
+The most important principle for dealing with communciation between models in Tungsten.js is this: communication up the model tree should be done via events, communication down the model tree should be done by calling methods on the models.  Events up, methods down.
+
+What does this mean?  Suppose there are two models in a collection that need to communicate with one another.  Instead of the communication occuring directly between these two sibling models, the communication is going to go through their nearest ancestor, which in this case would be the parent model of the collection.  To do this, one model will trigger an event, e.g., `this.trigger('someEvent', data);` and the parent model will listen to that event and call a method on the second model to communicate the necessary information.  It will find which model in the collection to call the method on likely via a `collection.findWhere(attrs)` call.  Ideally the event and method should pass through only the information necessary for the communication and no more.  This is a simple case of events up, methods down for communicating between models, but it scales to more complex cases as well.
```
Comment (danieluhl) [inline docs/patterns.md]: Can we put an example of the scoped listenTo in here `this.listenTo(this.model, 'event:object:property', fn);` I find people often don't know about or have trouble finding documentation on this.

Comment (danieluhl) [inline docs/patterns.md]: Or this could go down in the 'Trigger and listen to model, not view, events' section

File: docs/patterns.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+In the view's `postInitialize`, call `this.model.fetch()` (or another method that will make the ajax request and set the remaining part of the model).  This needs to be called from the _view_ `postInitialize`, not the model since the model is likely being instantiated before its attached to the view.  Once the data is set (in the callback from the ajax and set call), toggle the `loading` boolean on the model, switching to the standard template.
+
+## DOM event handlers
+
+The primary purpose of a view is responding to DOM events and calling methods on its model.  DOM events are all set in the `events` hash using either built in DOM events or one of the special events packaged with Tungsten.js.  A useful practice is to name all of the event handlers with the pattern "handleEventSelector".  So a plain `click` event would call the handler function `handleClick`, and a `change .js-foo` event would call the handler function `handleChangeFoo`.
```
Comment (danieluhl) [inline docs/patterns.md]: Can we add something about not adding listeners in the model postInitialize and why that's a bad idea

Comment (andrewrota) [inline docs/patterns.md]: I think this is handled, indirectly at least, in the deferred fetch section

File: docs/patterns.md (hunk 1 above)
Comment (andrewrota) [inline docs/patterns.md]: I'd rather reference new events documentation from #261

**Pull request merged**